### PR TITLE
connecting to mysql

### LINKED ISSUE #9: Implement user information function
## Summary

#1 `GET /users/:username`

This task is implementation of the user information function.
This function returns user information by JSON with 200 HTTP status code if a user accessed has valid JWT.
If a user doesn't have valid JWT, this returns empty JSON with 401 HTTP status code.

### PR DESCRIPTION
**Pull request by yamakenji24**
## implementation list
- [x] connect to mysql
- [x] encode jwt by bcrypt password
- [x] decode jwt and search user

### Mysql Connection
- connect to mysql in db.py
- connections requires some config data, which can be used from env or by default.

### Password encoding and JWT
- changed using fake information to DB data
- jwt is created by RS256, which key is generated from pem(secret)
  - directions to the pem file can be defined from env, or by default

#### demo
スクリーンショット 2020-11-08 3 20 56

### Authorization
- authorize user from jwt
- return user information if jwt is as expected
#### demo
スクリーンショット 2020-11-08 3 23 14

### CODE REVIEW

**Comment by yamakenji24** [comment]
## What i did in this PR
### Mysql Connection
- connect to mysql in db.py
- connections requires some config data, which can be used from env or by default.
-

File: .gitignore [hunk 1]
```diff
@@ -1 +1,3 @@
-key/*.pem
\ No newline at end of file
+key/*.pem
+__pycache__
+*/**/__pycache__
```
Comment (wakatakeru) [inline .gitignore]: recommend adding new line

File: server.py [hunk 2]
```diff
@@ -1,26 +1,28 @@
...
 
 @app.post("/register")
 async def register_user():
-  return create_user()
+  return user.create_user()
 
-@app.get("/users/{user_id}")
-async def get_info(user_id: str):
-  return fetch_user_info(user_id)
+@app.get("/users/me")
```
Comment (wakatakeru) [inline server.py]: change `/users/me` to `/my`

File: src/db.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  encoding="utf-8",
+  echo=True
+)
+
+session = scoped_session(
+  sessionmaker(autocommit=False, autoflush=False, bind=engine)
+)
+
+Base = declarative_base()
+Base.query = session.query_property()
```
Comment (wakatakeru) [inline src/db.py]: recommend adding new line

**Review by wakatakeru — CHANGES_REQUESTED** [review]
request changes

**Review by wakatakeru — APPROVED** [review]
LGTM!

**Pull request merged**